pylint/checkers/async.py

```diff
@@ -94,6 +98,9 @@ def visit_asyncwith(self, node: nodes.AsyncWith) -> None:
             )
 
 
-def register(linter):
-    """required method to auto register this checker"""
+def register(linter: "PyLinter") -> None:
+    """This required method auto registers the checker during initialization.
+
+    :param linter: The linter to register the checker to.
+    """
```

Comment:
Just my opinion, but after seeing this in every checker, I don't see much value in having a docstring from `register` aside from the examples in the documentation. It's unlikely someone will need to modify it once the checker is created.

My recommendation: Remove the docstring from all builtin checkers instead.

Reply:
This was my recommendation as I felt it was unproductive to remove docstrings that had already been written.
If (for whatever reason) we ever want to build documentation or anything from docstrings like we do in `astroid` it will be nice not to need to add them again. Besides, personally I don't mind duplication of docstrings as I find it helps attract contributors (and myself) to code. I find it better than having to look for `def register(` in the repository in the hope that some other similar function does have a docstring.

Reply:
IMO documentation should mainly be used  **public** classes and methods. For whom would you want to publish implementation details? At the very least, remove the `:param linter:` line. A one liner is more than enough. For everything more, one should read the documentation anyway.

Reply:
Hmm I don't fully agree. Considering the size and age of our codebase I think some internal documentation is often quite useful. Also because we don't really have a steady group of employees that work with the internal code, but an ever-changing group of outside contributors. Besides implementation details we would ideally also publish contributing details, of which documentation in functions such as these should function.

I agree this might be a bit overkill, but looking at the typing effort and the difficulty we sometimes face to see how and why certain internal functions behave in a certain way I believe internal documentation is also a good step towards easier contributions.

Reply:
It's already difficult enough to keep the public documentation up to date. In my experience no documentation is still better than outdated one. It's just one more thing that needs to be maintained, for just not enough benefit.

Reply:
What about keeping it in the doc examples, and removing it in the code ?

Reply:
> What about keeping it in the doc examples, and removing it in the code ?

👍🏻 Works for me.

Reply:
4a1c99a